src/osmf/js/osmf_flash.js

```diff
@@ -457,51 +469,38 @@ require("../../../html5-common/js/utils/constants.js");
       this.callToFlash("destroy");
 
       // Remove the element
-      $('#'+playerId).replaceWith('');
-      _flashVideoObject=null;
-
-      // return unbound object.
-      return JFlashBridge.unbind(playerId);
-    };
-
-    // Returns the SWF instance
-    this.swf = function () {
-      return JFlashBridge.getSWF(playerId);
+      $('#'+domId).replaceWith('');
+      _video=null;
+      $(_video).remove();
+      delete flashItems[this.id];
     };
 
     // Calls a Flash method
     this.callToFlash = function (data,dataObj) {
-      if(_flashVideoObject == undefined) { 
+      if(_video == undefined) { 
         javascriptCommandQueue.push([data,dataObj]);
       }
       else
-      {      
-        if (_flashVideoObject.sendToActionScript) {
+      {
+        if (_video.sendToActionScript) {
           dataObj = typeof dataObj != 'undefined' ? dataObj : "null";
-          return _flashVideoObject.sendToActionScript(data,dataObj);
+          return _video.sendToActionScript(data,dataObj, this.id);
         } 
-        else 
+        else
         {
```

Comment:
These should be cleaned up now for all lines ABOVE "SWFObject v2.2". SWFObject v2.2 is an external library we've pulled in and slightly modified, that is on the todo list to extract back out and develop around the portions we had to alter. So we've left it alone to keep that task manageable.